### PR TITLE
Feature/code refactoring

### PR DESCRIPTION
**Pull request by OSKyriienko**

### CODE REVIEW

File: src/components/dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -1,74 +1,66 @@
...
-
 const Dashboard = (props) => {
 
   const [loading, setLoading] = useState(true);
   const [visible, setVisible] = useState(true);
-  const [amountWS, setAmountWS] = useState(0);
   const [workspaces, setWorkspaces] = useState(null);
+  const [ws, setWs] = useState(null);
 
   useEffect(() => {
```
Comment (TuxujPes) [inline src/components/dashboard/Dashboard.jsx]: unnecessary empty function

File: src/components/dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -1,74 +1,66 @@
...
     await profileService.createWorkspaces(data);
     setVisible(false);
-  }  
+  }
 
   const onSubmitSelectWS = async (data) => {
     console.log(data);
     localStorage.setItem("currentWS", data.team);
-    history.push(`/dashboard/${localStorage.getItem("currentWS")}`);
+    // setWs(find(workspaces, { 'name': data.team }));
```
Comment (TuxujPes) [inline src/components/dashboard/Dashboard.jsx]: cleanup comments

File: src/components/dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -82,11 +74,25 @@ const Dashboard = (props) => {
             {handleSubmit}) => (
             <form onSubmit={handleSubmit}>
```
Comment (TuxujPes) [inline src/components/dashboard/Dashboard.jsx]: form nested in a form?

File: src/components/profileService/profileService.jsx [hunk 4]
```diff
@@ -23,19 +24,17 @@ class profileService {
       },
     })
 
-    const ws = await graphQLClient.request(query)
+    return graphQLClient.request(query)
     .then(data => {
```
Comment (TuxujPes) [inline src/components/profileService/profileService.jsx]: do not mix promises and async/await in a project.
follow async/await approach

File: src/components/dashboard/Dashboard.jsx (hunk 3 above)
Comment (OSKyriienko) [reply src/components/dashboard/Dashboard.jsx]: <form> inside the <Form>. It's a Final-form approach.

File: src/components/mainMenu/MainMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        <Button type="link"  onClick={() => history.push('/main/profile')}>Profile</Button>
+        <Button type="link"  onClick={() => history.push('/main/leaves')}>Leaves</Button>
+        <Button type="link"  onClick={() => history.push(`/main/workspaces`)}>Workspaces</Button>
+        <Button type="link"  onClick={() => history.push('/main/calendar')}>Team Calendar</Button>
+        <Button type="link"  onClick={() => history.push('/main/todo')}>Todo</Button>
+        <Button type="link"  onClick={() => authService.logout()}>Log out</Button>
+      </div>
+        <Switch>
+          <Route path="/main/workspaces" component={Workspaces} />
+          <Route path="/main/calendar" component={Calendar} />
```
Comment (marcusn90) [inline src/components/mainMenu/MainMenu.jsx]: Move route paths to constant and use in Route and history.push

File: src/services/profileService/profileService.jsx [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+        name: "${name}", 
+        description: "${description}",
+         members: ["${localStorage.getItem("email")}"]
+      ) {
+        ws{id}
+      }
+    }
+    `;
+
+    return await this.getGraphQLClient.request(query);
```
Comment (marcusn90) [inline src/services/profileService/profileService.jsx]: No need to use async-await here

File: src/components/calendar/Calendar.jsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    const ldata = holidays.filter(wsDay => wsDay.date === cellDate);
+    return (
+      <>
+        { !!ldata.length && <span>.</span> }
+      </>
+    );
+  }
+
+  const showEvents = () => {
+    console.log(listdata);
```
Comment (TuxujPes) [inline src/components/calendar/Calendar.jsx]: clean up `console.log`

File: src/services/authService/authService.jsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+      localStorage.setItem("email", login);
+      history.push('/main/profile');
+    }
+    catch(error) {
+      console.log(error);
+      history.push('/auth/signup');
+    };
+  }
+
+  getStatus = () => {
```
Comment (TuxujPes) [inline src/services/authService/authService.jsx]: get status or get token?

File: src/services/profileService/profileService.jsx [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+import { find } from 'lodash';
+
+// export const endpoint = "http://nmarchuk.pythonanywhere.com/graphql";
+export const endpoint = "http://localhost:3344/graphql";
+
+class profileService {
+
+  workspaces = null;
+
+  get getGraphQLClient() {
```
Comment (TuxujPes) [inline src/services/profileService/profileService.jsx]: create your own service for graphql requests with `Authorization` in mind.

File: package.json [hunk 10]
```diff
@@ -35,5 +35,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "moment": "^2.24.0"
```
[the remark this answers is not in the document]
Comment (OSKyriienko) [reply package.json]: moment is already used for date formatting and was moved to dependencies

File: src/components/calendar/Calendar.jsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    (async() => {
+      const workspaces = await profileService.fetchMyWorkspaces();
+      // setWorkspaces(workspaces);
+      console.log(workspaces);
+      const id = find(workspaces, { 'name': localStorage.getItem("currentWs") }).id;
+      const holidayDays = await profileService.getHolidayData(id);
+      setHolidays(holidayDays.workspaceDates);
+      setLoading(false);
+    })();
+  }, [loading]);
```
[the remark this answers is not in the document]
Comment (OSKyriienko) [reply src/components/calendar/Calendar.jsx]: No, it was an error, that caused double call. Now it fixed.

File: src/components/calendar/Calendar.jsx [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+    (async () => {
+      try {
+        const [holidays, vacations] = await Promise.all([
+          profileService.getHolidayData(profileService.getWs.id),
+          profileService.getVacationDays(profileService.getWs.id)
+        ]);
+        setHolidays(holidays.workspaceDates);
+        setVacations(vacations.teamCalendar);
+      } catch (error) {
+        throw error;
```
Comment (marcusn90) [inline src/components/calendar/Calendar.jsx]: why do we need trycatch if you re-throw?

**Pull request merged**